scripts/Muon/GUI/Common/fitting_widgets/model_fitting/model_fitting_model.py

```diff
@@ -0,0 +1,194 @@
+            if parameter_name not in y_parameter_error_names:
+                self.fitting_context.y_parameter_errors[parameter_name] = [0.0] * number_of_data_points
+
+    def _create_workspace_group_to_store_combination_workspaces(self) -> WorkspaceGroup:
+        """Return the Workspace Group used to store the different parameter combination matrix workspaces."""
+        group_name = self.parameter_combination_group_name()
+        if group_name is None:
+            return None
+
+        if check_if_workspace_exist(group_name):
+            workspace_group = retrieve_ws(group_name)
+        else:
+            workspace_group = WorkspaceGroup()
+            add_ws_to_ads(group_name, workspace_group)
+        return workspace_group
+
+    def _create_matrix_workspaces_for_parameter_combinations(self, workspace_group: WorkspaceGroup) -> list:
+        """Creates a MatrixWorkspace for each parameter combination. These are the workspaces that will be fitted."""
+        workspace_names = []
+        for x_parameter_name in self.fitting_context.x_parameters.keys():
+            for y_parameter_name in self.fitting_context.y_parameters.keys():
+                x_values = self._convert_str_column_values_to_int(x_parameter_name, self.fitting_context.x_parameters)
+                y_values = self._convert_str_column_values_to_int(y_parameter_name, self.fitting_context.y_parameters)
+                y_errors = self.fitting_context.y_parameter_errors[y_parameter_name]
+
+                # Sort the data based on the x_values being in ascending order
+                x_values, y_values, y_errors = zip(*sorted(zip(x_values, y_values, y_errors)))
+
+                output_name = self.parameter_combination_workspace_name(x_parameter_name, y_parameter_name)
+                if not self._parameter_combination_workspace_exists(output_name, x_values, y_values, y_errors):
+                    CreateWorkspace(DataX=x_values, DataY=y_values, DataE=y_errors, OutputWorkspace=output_name)
+                    workspace_group.add(output_name)
+
+                workspace_names.append(output_name)
+
+        return workspace_names
+
+    def _parameter_combination_workspace_exists(self, workspace_name: str, x_values: list, y_values: list, y_errors: list) -> bool:
+        """Returns true if a parameter combination workspace exists and contains the same data."""
+        if check_if_workspace_exist(workspace_name):
+            workspace = retrieve_ws(workspace_name)
+            return self._data_is_equal(workspace.dataX(0), x_values) \
+                   and self._data_is_equal(workspace.dataY(0), y_values) \
+                   and self._data_is_equal(workspace.dataE(0), y_errors)
+        return False
+
+    @staticmethod
+    def _data_is_equal(list1: list, list2: list) -> bool:
+        """Returns true if the two lists containing x, y or error data are equal to five decimal places."""
+        return all([f"{i:.5f}" == f"{j:.5f}" for i, j in zip(list1, list2)])
```

Comment:
I have seen something like this elsewhere. Is it worth making the 5 a variable, so it can be easily changed?